CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/icon/IconDataNGTest.java

```diff
@@ -0,0 +1,465 @@
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ * Test class for IconData.
+ *
+ * @author sol695510
+ */
+public class IconDataNGTest {
+
+    private static MockedStatic<IconData> iconDataStaticMock;
+    private static MockedStatic<ImageIO> imageIOStaticMock;
+    private static InputStream inputStreamMock;
+    private static BufferedImage bufferedImageMock;
+    private static IOException IOExceptionMock;
+
+    public IconDataNGTest() {
+    }
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+        iconDataStaticMock = Mockito.mockStatic(IconData.class);
+        imageIOStaticMock = Mockito.mockStatic(ImageIO.class);
+        inputStreamMock = Mockito.mock(InputStream.class);
+        bufferedImageMock = Mockito.mock(BufferedImage.class);
+        IOExceptionMock = Mockito.mock(IOException.class);
+    }
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+        iconDataStaticMock.close();
+        imageIOStaticMock.close();
+    }
+
+    @BeforeMethod
+    public void setUpMethod() throws Exception {
+        iconDataStaticMock.reset();
+        imageIOStaticMock.reset();
+    }
+
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+    }
+
+    /**
+     * Test of getData method, of class IconData.
+     */
+    @Test
+    public void testGetData() {
+        System.out.println("testGetData");
+
+        // When niether size is the default (256) or color is null.
+        final IconData instance1 = spy(new IconDataImpl());
+
+        final int size1 = 256;
+        final Color color1 = new Color(1, 1, 1);
+
+        final byte[] expResult1 = new byte[0];
+        final byte[] result1 = instance1.getData(size1, color1);
+
+        verify(instance1, times(1)).createData(Mockito.eq(size1), Mockito.eq(color1));
+        assertEquals(result1, expResult1);
+
+        // When color is null.
+        final IconData instance2 = spy(new IconDataImpl());
+
+        final int size2 = 0;
+        final Color color2 = null;
+
+        final byte[] expResult2 = new byte[0];
+        final byte[] result2 = instance2.getData(size2, color2);
+
+        verify(instance2, times(1)).createData(Mockito.eq(size2), Mockito.eq(color2));
+        assertEquals(result2, expResult2);
+
+        // When size is the default (256).
+        final int size3 = 256;
+        final Color color3 = null;
+
+        // When data is null.
+        final IconData instance3 = spy(new IconDataImpl());
+        instance3.setData(null);
```

Comment:
I think you should only need one of these. That is either creating a new instance every time, or adding the setter to be able to set back to null. Personally I would prefer the former (since I'm guessing there is a reason why there isn't a setter in place judging by the code) but even then I think if you find yourself needing to create more than one instance in a unit test then I would possibly look at splitting the tests up